http/client/src/main/java/org/wildfly/security/http/client/mechanism/digest/util/DigestHttpMechanismUtil.java

```diff
@@ -0,0 +1,150 @@
+        if(opaque != null) requestAuthHeader.append("uri=\"").append(uriPath).append("\", ");
+        requestAuthHeader.append("qop=\"").append(qop).append("\", ");
+        requestAuthHeader.append("nc=\"").append(nc).append("\", ");
+        requestAuthHeader.append("cnonce=\"").append(cnonce).append("\", ");
+        requestAuthHeader.append("response=\"").append(resp).append("\", ");
+        requestAuthHeader.append("opaque=\"").append(opaque).append("\", ");
+        requestAuthHeader.append("algorithm=").append(algorithm);
+
+        Map<String, String> headers = new HashMap<>();
+        headers.put(ElytronHttpClientConstants.AUTHORIZATION, requestAuthHeader.toString());
+        HttpRequest request = ElytronHttpClientRequestBuilder.buildRequest(httpRequest, headers);
+        int updateNonceCount = Integer.parseInt(authParams.getOrDefault("nc", "0")) + 1;
+        authParams.put("nc", String.valueOf(updateNonceCount));
+        return request;
+    }
+
+    private static Map<String, String> updateAuthParams(String authHeader) throws AuthenticationMechanismException {
+        byte[] rawHeader = authHeader.substring(ElytronHttpClientConstants.CHALLENGE_PREFIX.length()).getBytes(UTF_8);
+        HashMap<String, byte[]> authval = parseResponse(rawHeader, UTF_8, false, log);
+        Map<String, String> authParams = new HashMap<>();
+        for (String headerKey : authval.keySet()) {
+            String headerValue = new String(authval.get(headerKey), UTF_8);
+            authParams.put(headerKey, headerValue);
+        }
+        return authParams;
+    }
+
+    private static String computeDigestWithoutQop(String uri, String nonce, String username, String
+            password, String algorithm, String realm, String method) {
+        String A1, HashA1, A2, HashA2;
+        A1 = username + ":" + realm + ":" + password;
+        HashA1 = generateHash(A1, algorithm);
+        A2 = method + ":" + uri;
+        HashA2 = generateHash(A2, algorithm);
+        String combo, finalHash;
+        combo = HashA1 + ":" + nonce + ":" + HashA2;
+        finalHash = generateHash(combo, algorithm);
+        return finalHash;
+    }
+
+    private static String computeDigestWithQop(String uri, String nonce, String cnonce, String nc, String
+            username, String password, String algorithm, String realm, String qop, String method) {
+
+        String A1, HashA1, A2, HashA2;
+        A1 = username + ":" + realm + ":" + password;
+        HashA1 = generateHash(A1, algorithm);
+        A2 = method + ":" + uri;
+        HashA2 = generateHash(A2, algorithm);
+
+        String combo, finalHash;
+        combo = HashA1 + ":" + nonce + ":" + nc + ":" + cnonce + ":" + qop + ":" + HashA2;
+        finalHash = generateHash(combo, algorithm);
+        return finalHash;
+    }
+
+    private static String generateHash(String value, String algorithm) {
+        try {
+            MessageDigest messageDigest;
+            if (algorithm.equals("SHA-512-256")) {
```

Comment:
@keshav-725 Just aminor, why is the SHA-512-256 treated differently here than other algorithms? can we used the WildFlyElytronDigestProvider for all algorithms instead?